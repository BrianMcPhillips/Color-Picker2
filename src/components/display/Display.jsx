import React from 'react';
import PropTypes from 'prop-types';

const Display = ({ bgColor, fgColor, text }) => (
  <div data-testid="display" style={{ 
    backgroundColor: bgColor,
    color: fgColor,
    width: '500px',
    height: '500px'
  }}>
    {text}
  </div>
);

Display.propTypes = {
  bgColor: PropTypes.string.isRequired,
  fgColor: PropTypes.string.isRequired,
  text: PropTypes.string.isRequired,
};

export default Display;
