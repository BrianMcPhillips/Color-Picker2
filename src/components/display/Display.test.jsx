import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Display from './Display';

describe('Display component', () => {
  afterEach(() => cleanup());
  it('renders Display', () => {
    const { asFragment } = render(<Display
      bgColor="#FF0000"
      fgColor="00FF00"
      text="Hiiiii" />);
    expect(asFragment()).toMatchSnapshot();
  });
});
