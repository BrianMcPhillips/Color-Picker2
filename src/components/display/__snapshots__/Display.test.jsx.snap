// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display component renders Display 1`] = `
<DocumentFragment>
  <div
    data-testid="display"
    style="background-color: rgb(255, 0, 0); width: 500px; height: 500px;"
  >
    Hiiiii
  </div>
</DocumentFragment>
`;
